apt-hook/20apt-esm-hook.conf

```diff
@@ -1,3 +1,7 @@
+APT::Update::Pre-Invoke {
+	"[ ! -f /usr/lib/ubuntu-advantage/apt_update_nonblocking_hook.py ] || flock -n -E 0 /var/lib/ubuntu-advantage/apt-update-lock /usr/lib/ubuntu-advantage/apt_update_nonblocking_hook.py &";
+};
+
```

Comment:
Well I guess that works, but it might be nicer to just use `fnctl.flock()` or `fcntl.lockf()` [that's the type of lock apt and dpkg use] in the script itself?

Might be worth instead of forking a background process here to `systemd-run` it `if [ -e /run/systemd/system ]` or stuff it into a .service you launch (with a timeout even).

That seems a bit more resilient and easier to investigate.

Reply:
Thanks for the quick feedback! That all makes sense. I'll move it to a .service file and just call `systemctl start apt-news.service` here.

And yep good point It'll be easier to follow if the lock check is in the script itself. I'll make that change as well

Reply:
If you don't care about news inside docker and chroots (systems without systemd), you don't need to worry about locking at all if it's a service.

But I don't know if you do

Reply:
Ah right! I think it's fair to not worry about this feature in non-systemd environments

Reply:
Updated so moving the conversation [here](https://github.com/canonical/ubuntu-advantage-client/pull/2338#discussion_r1044786676) for any follow ups. Resolving this one.